# GrandSTR

Estimation of repeat counts of STR from long-read sequencing data.
It's a software for genotyping of known short tandem repeats(STR).


## Dependencies

- python: 3.6 or higher
- pysam: 0.16.0 or higher
- sklearn: 0.24.1 or higher
- hmmlearn: 0.2.5 or higher
- edlib: 1.2.6 or higher

Dependencies for install:
- cython: 0.29.21 or higher


## Install

build align.so, GrandSTR_lib.so, utils_lib.so, deal_str.so:

```bash
python setup.py build_ext -i
```

## Usage & Examples

Input bam file is alignment file of read sequences aligned to reference sequences, which is typically generated by minimap2 with parameters "-ax asm10 --MD -Y -L --secondary=no" for hifi reads, or "-ax map-ont --MD -Y -L --secondary=no" for ONT reads. Input pa file is comma seperated information file, including coordinates of STR regions in reference, and repeat unit sequence. Input fasta file is reference genome fasta file. 

For small amount of input STRs provided in pa file, add "-em 0" parameter to GrandSTR.py program. For example:
```bash
samtools index hifi.sorted.bam
samtools faidx hg19.fasta
GrandSTR test1.pa out1 -rf hg19.fasta -bf hifi.sorted.bam -em 0 -rt hifi
```

For large amount of input STRs provided in pa file, add "-em 1" parameter to GrandSTR.py program. For example:
```bash
samtools index hifi.sorted.bam
GrandSTR test1.pa out1 -rf hg19.fasta -bf hifi.sorted.bam -em 1 -rt hifi
```
